nomad/leader.go

```diff
@@ -532,8 +532,10 @@ func (s *Server) reapFailedEvaluations(stopCh chan struct{}) {
 			// due to the fairly large backoff.
 			followupEvalWait := s.config.EvalFailedFollowupBaselineDelay +
 				time.Duration(rand.Int63n(int64(s.config.EvalFailedFollowupDelayRange)))
+
 			followupEval := eval.CreateFailedFollowUpEval(followupEvalWait)
 			updateEval.NextEval = followupEval.ID
+			updateEval.ModifyTime = time.Now().UTC().UnixNano()
```

Comment:
Also, on a leadership transition, this could create a situation where `CreateTime` is later than `ModifyTime`, because the new leader might have a slow clock. At this scope that's probably fine, but there could be an argument for at least a naive lamport clock, which would go

```now := time.Now().UTC().UnixNano()
if now < e.CreateTime {
   e.ModifyTime = e.CreateTime + 1
} else {
  e.ModifyTime = now
}```

Reply:
this would support future dependence on the timestamps to create an isUpdated condition, and to avoid surprise / confusion in log messages

Reply:
A similar constraint could exist between evals that are related, like if it's a problem for a blocked eval's `CreatedTime` to be earlier than the `CreatedTime` of the eval referenced by it's `PreviousEval`

Reply:
Done